src/components/Navbar.tsx

```diff
@@ -1 +1,75 @@
-export const Navbar = () => <>Nav</>;
+import { useState } from 'react';
+
+import { useLogin } from 'hooks/context/user/userContext';
+import { userLogoutModule } from 'hooks/context/user/userContextModule';
+import { useVideoList } from 'hooks/context/videoContext';
+import {
+  BsFillMoonStarsFill,
+  BsSunFill,
+} from 'react-icons/bs';
+import {
+  Link,
+  useLocation,
+  useNavigate,
+} from 'react-router-dom';
+
+export const Navbar = () => {
+  const { searchVideoList } = useVideoList();
+  const [theme, setTheme] = useState("dark");
+  const [searchVideo, setSearchVideo] = useState("");
+  const { pathname } = useLocation();
+  const navigate = useNavigate();
+  function handleSearchVideo(searchVideo: string) {
+    if (pathname !== "/") navigate("/");
+    searchVideoList(searchVideo);
+  }
+  const { loginUser, userDispatch, isAuth } = useLogin();
+  return (
+    <nav className="bg-gacol p-4 text-wcol flex">
+      <div className="mr-auto">
+        {" "}
+        {isAuth() && (
+          <Link to="profile">
+            <div className="rounded-full bg-lcol cursor-pointer text-dcol w-[2.7ex] text-center font-bold ">
+              {loginUser.fname[0].toUpperCase()}
+            </div>
+          </Link>
+        )}
+      </div>
+      <div className="mx-auto flex justify-center">
+        <input
+          type="text"
+          className=" w-3/4 bg-lcol border-2 border-gacol rounded-md outline-none px-1"
+          value={searchVideo}
+          onChange={(e) => {
+            setSearchVideo(e.currentTarget.value);
+            handleSearchVideo(searchVideo);
+          }}
+          placeholder="Search"
+        />{" "}
+      </div>
+      <div className="flex gap-1 ml-auto">
+        <button
+          onClick={() => {
+            document.body.classList.toggle("theme_light");
```

Comment:
Instead of using vanilla js , 
the states can be kept in themecontext which could be used as classnames and hence avoiding the use of js

Reply:
That's an opinionated way of doing it. Before using this way I had a good rundown of two-three different ways I can tackle it. while there is one way I can still avoid it by " adding this class name to the App wrapper itself "adding it to the body is the standard way so I followed it. 
In the part where I am using vanilla JS, I find it okay as there is no way I can access the body itself and I need to set global variables.
A class-based theme is a good idea but maintaining a whole context for just theme switching is probably an overkill. I know it provides a lot of customizability but at the same time, it is not in line with the idea of a single point of control. 
Let me know If we can discuss it more if it is needed. For now, I'll keep it this way.